### PR TITLE
update craft-archives and support apt pinning

### PR DESCRIPTION
**Pull request by tigarmo**
This commit updates craft-archives to 0.0.3, which restructures the
key-handling implementation and support repo pinning.

CRAFT-1646

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def recursive_data_files(directory, install_directory):
 install_requires = [
     "attrs",
     "click",
-    "craft-archives==0.0.1",
+    "craft-archives==0.0.3",
```
Comment (tigarmo) [inline setup.py:99]: does anyone know why we pin the version here too?

File: tests/spread/core22/package-repositories/task.yaml [hunk 2]
```diff
@@ -5,15 +5,14 @@ environment:
...
   SNAP/test_apt_ppa: test-apt-ppa
+  SNAP/test_pin: test-pin
   SNAPCRAFT_BUILD_ENVIRONMENT: ""
 
 restore: |
   cd "$SNAP"
   rm -f ./*.snap
   snapcraft clean
   snapcraft clean --destructive-mode
-  ls -l /etc/apt/sources.list.d/
```
Comment (tigarmo) [inline tests/spread/core22/package-repositories/task.yaml:15]: removed this because tests/spread/tools/restore.sh handles this per-test already.

**Review by lengau — APPROVED** [review]
Thanks, this is excellent!

**Review by sergiusens — APPROVED** [review]
very nice!

**Pull request merged**